app/src/js/components/Logs/viewer.js

```diff
@@ -16,140 +18,160 @@ const noLogs = {
 };
 // const twoDays = 2 * 1000 * 60 * 60 * 24;
 
-const statusOptions = [
-  'All',
-  'Debug',
-  'Info',
-  'Warn',
-  'Error',
-  'Fatal',
-  'Trace'
-];
+// Use a Map to preserve order of entries.  Otherwise, when using a plain
+// object, during iteration over the keys (or entries), "numeric" keys are
+// yielded in numeric order, not in insertion order (Gah!).
+const logLevels = new Map([
+  ['[10 TO 60]', 'All'],
+  ['60', 'Fatal'],
+  ['50', 'Error'],
+  ['40', 'Warn'],
+  ['30', 'Info'],
+  ['20', 'Debug'],
+  ['10', 'Trace'],
+]);
+
+/**
+ * Returns the name of the specified "numeric" log level, or the specified
+ * level converted to a string if no such level is defined.
+ *
+ * @param {number | string} level "numeric" log level to get the name of
+ * @returns {string} name of the specified log level or the level converted to
+ *    a string if no such level is defined
+ */
+const logLevelName = (level) => (logLevels.get(`${level}`) || `${level}`);
 
 class LogViewer extends React.Component {
-  constructor () {
-    super();
+  constructor(props) {
+    super(props);
+
     this.displayName = 'LogViewer';
-    this.setSearch = this.setSearch.bind(this);
-    this.setSearchLevel = this.setSearchLevel.bind(this);
+
     this.query = this.query.bind(this);
+    this.cancelInterval = () => { };
+
     this.state = {
-      level: 'All',
-      search: ''
+      level: logLevels.keys().next().value,
+      search: '',
     };
   }
 
-  componentDidMount () {
+  componentDidMount() {
     this.query();
   }
 
-  componentDidUpdate (prevProps) {
-    if (JSON.stringify(this.props.query) !== JSON.stringify(prevProps.query)) {
```

Comment:
This line is deleted.  It seems that this will create unnecessary queries and page updates.

Reply:
No, because I also removed the call to `this.query`, which is what this condition was guarding.